data/core.yaml

```diff
@@ -499,6 +499,8 @@ en:
           one: "Split a feature."
           other: "Split {n} features."
       not_eligible: Lines can't be split at their beginning or end.
+      parent_incomplete: This line cannot be split because it is part of a larger relation which has only been partially loaded. Make sure that all ways connected to this way are present on the map.
```

Comment:
Yeah, this one was tricky to write a concise and understandable description for.

And you're right: the entities don't need to be actually _visible_ in the viewport. I thought about automatically downloading the missing prev/next members in such cases (it's really only up to two members, which is not a lot), but unfortunately the edit operations don't support async workflows at the moment (which would be required here to do the downloading).

> How about adding a button to the top or bottom of the relation member list to load the full relation on demand?

That sounds useful. But maybe in a different issue to not overload this one too much.

> full sort on demand, like JOSM has

I thought about that as well, but left it for another day/issue/pr. :wink: I think such a feature should come with some kind of (very) basic visualization of the order too.

Reply:
iD already has a good enough visualization for this basic purpose (hovering over members when zoomed to fit). I just wouldn’t want the outcome of this genuine improvement to be that people still tell new mappers to only map route relations in JOSM because it doesn’t clean up after new or existing scrambling.

I think #5420 is more tractable than #6656 because we won’t wastefully spam the API for unwanted data, and because the code basically already exists. Some other messages warn about a way not being fully visible, but at least there’s a button and keyboard shortcut for that.

If you’d like to keep the scope to a minimum, that’s fine, but then I have no idea how to make this message genuinely helpful. It might turn out to be easier to implement one of the two buttons. Is the new implementation still capable of a full relation sort, or would that functionality have to be reimplemented?

Reply:
d7111e2 adds a button to do the `/full` download of a particular relation.